app/src/main/java/org/fossasia/susi/ai/data/device/DeviceServiceImpl.kt

```diff
@@ -76,4 +77,23 @@ class DeviceServiceImpl : DeviceService {
             }
         })
     }
+
+    override fun submitRoomDetails(roomName: String, listener: IDeviceModel.onSendRoomDetails) {
+        deviceApi.roomDetails(roomName).enqueue(object : Callback<AddRoomResponse> {
+            override fun onFailure(call: Call<AddRoomResponse>, t: Throwable) {
+                if (t?.localizedMessage != null) {
+                    Timber.e(t.localizedMessage)
+                } else {
+                    Timber.e(t, "An error occurred")
+                }
+                Timber.e(t, "Error in WiFi : " + call.toString())
```

Comment:
Okay, as you can see there are two timer statements which will execute, can you combine both?

Reply:
Actually here we have 3 timber statements. The first 2 depends on the localizedMessage, the 3rd one is common. So, I don't think it would be a good idea to display the messages collectively. It's better to show them separately for better help in debugging and keeping track of the flow.

Reply:
But throwable print twice, then change it to a debug statement and remove throwable.

Reply:
Statement in else block is useless.

Reply:
Removed them. Also removed from other functions present in the file.